Add community values, improve code of conduct (#1655)
open-telemetry/community: code-of-conduct.md

# OpenTelemetry Community Code of Conduct

The OpenTelemetry project aims to be a welcoming place where new and existing
members feel safe to respectfully share their opinions and disagreements. We
want to attract a diverse group of people to collaborate with us, which means
acknowledging that people come from different backgrounds and cultures. 

> We are bound by the [CNCF Code of
> Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md) and
> covered by that in more serious violations and prefer the reporting of
> violations to be made to us as outlined below.

Reviewer: "covered by that in more serious violations" reads weirdly to me.

```suggestion
This project is bound by and abides by the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md).
Conflicts should be resolved respectfully between the involved parties, however
we understand that it may not always be possible without outside help or mediation.
When needed, any community member can ask for help or report a Code of Conduct
violation to the Governance Committee as outlined below.
The project prefers that violations to be reported first to the Governance Committee
so that they may be resolved internally, however, in extreme cases code of conduct
violations should be reported to the CNCF as outlined in the
[CNCF Incident Resolution Procedures](https://github.com/cncf/foundation/blob/main/code-of-conduct/coc-incident-resolution-procedures.md).
Anyone who is unsure if a violation should be reported to the project or the CNCF should
refer to the [CNCF CoC Jurisdiction Policy](https://github.com/cncf/foundation/blob/main/code-of-conduct/coc-committee-jurisdiction-policy.md).
```

## Reporting violations

If you have seen or experienced unacceptable behavior or anything that would
make our community less welcoming, please speak up! 

The Governance Committee is the right place to bring your concerns and can be
reached via cncf-opentelemetry-governance@lists.cncf.io. You can also get in

> touch with any committee member if you feel safer this way: any one of us can

Author: I'm marking this as resolved.  The only 👍🏽 missing is @alolita's, but I do think she expressed her consent in other channels (GC Slack channel?)

> touch with any committee member if you feel safer this way: any one of us can

Reviewer: I think it may be a good idea to provide a more specific example and be more direct about this. I'd take a look at the CNCF's CoC reporting procedure for inspiration https://github.com/cncf/foundation/blob/main/code-of-conduct/coc-incident-resolution-procedures.md#submit-in-writing

We may want to just take their wording almost exactly substituting our GC for their CoC Committee

> If you do not want your report to be received by all members of the CoC Committee, either because you want to submit a report anonymously or because one of the CoC Committee members has a conflict of interest, you may send your report directly to any individual member

Author: That doc is great. I reorganized the final part of our code of conduct with parts from that. Thank you for the hint! Can you please take another look?

> receive your report and will bring it to the entire group for investigation,

Reviewer: I don't think all reports should always be brought to the entire group and I wouldn't promise that here. It may prevent someone from raising a complaint against a GC member.

Author: Agreed, I replaced "will" by "may":

> any one of us can receive your report and may bring it to the entire group for investigation

anonymously if you prefer. The [current list of
members](https://github.com/open-telemetry/community/blob/main/community-members.md#governance-committee)
can be found on our community repository, and you can ping any of them directly
on [CNCF’s Slack](https://slack.cncf.io). The Governance Committee should be
diverse enough to conduct a neutral investigation having the best interests of
the project in mind at all times, and the results of the investigation will be
given back to the reporter in a timely fashion as an official position of the
GC. If you are unhappy with the outcome, you can escalate that to the CNCF via
conduct@cncf.io. Feel free to contact the CNCF directly if you have reasons to
believe that the board will not conduct a fair investigation.

When the GC determines that a code of conduct violation happened, it will take
the actions deemed necessary and appropriate for the event. In more serious

> cases, the community member might be invited to depart from our communities,

Reviewer 2: ```suggestion
cases, the community member might be requested to depart from our communities,
```

# OpenTelemetry Community Code of Conduct

The OpenTelemetry project aims to be a welcoming place where new and existing
members feel safe to respectfully share their opinions and disagreements. We
want to attract a diverse group of people to collaborate with us, which means
acknowledging that people come from different backgrounds and cultures. 

This project is bound by and abides by the [CNCF Code of

> Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md).

Reviewer 3: ```suggestion
Conduct](https://github.com/cncf/foundation/blob/main/code-of-conduct.md).
```

Conflicts should be resolved respectfully between the involved parties, however
we understand that it may not always be possible without outside help or
mediation. When needed, any community member can ask for help or report a Code
of Conduct violation to the Governance Committee as outlined below. The project
prefers that violations to be reported first to the Governance Committee so that
they may be resolved internally, however, in extreme cases code of conduct
violations should be reported to the CNCF as outlined in the [CNCF Incident
Resolution
Procedures](https://github.com/cncf/foundation/blob/main/code-of-conduct/coc-incident-resolution-procedures.md).
Anyone who is unsure if a violation should be reported to the project or the
CNCF should refer to the [CNCF CoC Jurisdiction
Policy](https://github.com/cncf/foundation/blob/main/code-of-conduct/coc-committee-jurisdiction-policy.md).

## Reporting violations

If you have seen or experienced unacceptable behavior or anything that would
make our community less welcoming, please speak up!

> When the Governing Committee (GC) determines that a code of conduct violation

Reviewer 3: ```suggestion
When the Governance Committee (GC) determines that a code of conduct violation
```

happened, it will take the actions deemed necessary and appropriate for the
event. In more serious cases, the community member might be requested to depart
from our communities, giving up any roles they may possess, such as the
maintainership of a SIG. In extreme cases, the GC can force the departure of the
member.

## Submit in writing

To report a violation in writing, please email
cncf-opentelemetry-governance@lists.cncf.io, which goes to all GC members. If
you do not want your report to be received by all GC members, either because you
want to submit a report anonymously or because one of the GC members has a
conflict of interest, you may send your report directly to any individual GC
member. The [current list of
members](https://github.com/open-telemetry/community/blob/main/community-members.md#governance-committee)
can be found on our community repository, and you can ping any of them directly
on [CNCF’s Slack](https://slack.cncf.io)

## Submit in spoken conversation

If you prefer to report the violation in a spoken conversation, you may request
a telephone conversation or virtual meeting with a GC member.

# How to report anonymously

> If you desire to submit a report anonymously, please send a message directly to

Reviewer 4: Nit: I am not sure we need to refer anonymous reporting in this paragraph. Contacting an individual GC member can definitely be more desirable in some cases, so perhaps we just specify the mechanics of how to do it without necessarily trying it to the idea of anonymity. Especially since this paragraph does not say how the contacting member should specify in their request that they want to remain anonymous.

Author: During a GC call, we thought the following doc would be a good source for us to adapt, and it's where this section comes from: https://github.com/cncf/foundation/blob/main/code-of-conduct/coc-incident-resolution-procedures.md#how-to-submit-a-report 

Personally, I like the option of having this explicitly listed here, even though most of it is already mentioned elsewhere (how to find GC members, how to report, ...)

> Especially since this paragraph does not say how the contacting member should specify in their request that they want to remain anonymous.

Not sure I understood this part. "Let them know you would like to report anonymously" should suffice here, I believe. Or do you mean that people might want to remain completely anonymous even to the party they are reporting the issue to?

Author: Tracked here: https://github.com/open-telemetry/community/issues/1684
